### PR TITLE
Added portable and failure-domain label to the OSD deployment

### PR DESCRIPTION
**Pull request by rohan47**
Signed-off-by: rohan47 <rohanrgupta1996@gmail.com>

**Description of your changes:**

Added portable and failure-domain label to the OSD deployment.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

### CODE REVIEW

File: pkg/operator/ceph/cluster/osd/spec.go [hunk 1]
```diff
@@ -290,6 +291,8 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo) (*apps.Dep
 				k8sutil.AppAttr:     appName,
 				k8sutil.ClusterAttr: c.Namespace,
 				osdLabelKey:         fmt.Sprintf("%d", osd.ID),
+				failureDomainKey:    failureDomainValue,
```
Comment (travisn) [inline pkg/operator/ceph/cluster/osd/spec.go]: How about factoring out a `getLabels()` method so we can apply these to both the deployment and pod template spec (line 312) consistently?

File: pkg/operator/ceph/cluster/osd/spec.go [hunk 2]
```diff
@@ -354,6 +357,8 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo) (*apps.Dep
 	} else {
 		deployment.Spec.Template.Spec.InitContainers = append(deployment.Spec.Template.Spec.InitContainers, c.getPVCInitContainer(osdProps.pvc))
 		k8sutil.AddLabelToDeployement(OSDOverPVCLabelKey, osdProps.pvc.ClaimName, deployment)
+		k8sutil.AddLabelToDeployement(failureDomainKey, failureDomainValue, deployment)
```
Comment (travisn) [inline pkg/operator/ceph/cluster/osd/spec.go]: Let's always add these labels to the deployment and pod. If not using a pvc, we would just expect `portable` to be false and the failureDomain to be the host name.

File: pkg/operator/ceph/cluster/osd/spec.go (hunk 1 above)
Comment (rohan47) [reply pkg/operator/ceph/cluster/osd/spec.go]: done

File: pkg/operator/ceph/cluster/osd/spec.go (hunk 2 above)
Comment (rohan47) [reply pkg/operator/ceph/cluster/osd/spec.go]: done

File: pkg/operator/ceph/cluster/osd/spec.go [hunk 3]
```diff
@@ -305,12 +302,8 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo) (*apps.Dep
...
 			Template: v1.PodTemplateSpec{
 				ObjectMeta: metav1.ObjectMeta{
-					Name: appName,
-					Labels: map[string]string{
-						k8sutil.AppAttr:     appName,
-						k8sutil.ClusterAttr: c.Namespace,
-						osdLabelKey:         fmt.Sprintf("%d", osd.ID),
-					},
+					Name:   appName,
+					Labels: c.getOSDLabels(osd.ID, failureDomainValue, strconv.FormatBool(osdProps.portable)),
```
Comment (travisn) [inline pkg/operator/ceph/cluster/osd/spec.go]: how about passing a bool and then format inside the getOSDLabels() method?

Comment (rohan47) [reply pkg/operator/ceph/cluster/osd/spec.go]: done

**Comment by leseb** [comment]
It would have been nice to add this to the release note.

**Comment by travisn** [comment]
Yeah, this will be documented and more fully implemented with a follow-up PR for #3648. This was just a first change to add labels to unblock the PDB work.

**Pull request merged**